WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -166,6 +169,26 @@ public void Spend(SmartCoin spentCoin, SmartTransaction tx)
 			{
 				InvalidateSnapshot = true;
 				SpentCoins.Add(spentCoin);
+
+				var newCoinSet = new HashSet<SmartCoin> { spentCoin };
+
+				// If we don't succeed to add a new entry to the dictionary.
+				if (!SpentCoinsByOutPoint.TryAdd(spentCoin.Outpoint, newCoinSet))
```

Comment:
I am not sure to understand this change but it seems to me that even if the transaction is undone, the information in this collections is not updated.